src/background.js

```diff
@@ -36,6 +36,9 @@ class ProviderDynamicQuickSuggest extends UrlbarProvider {
   }
 
   async isActive(queryContext) {
+    if (!(await api.suggestionsEnabled(queryContext.isPrivate))) {
+      return false;
+    }
     this.matchedResult = await api.matchSearchTerm(queryContext.searchString);
```

Comment:
After calling async code in `isActive` or `startQuery`, you'll want to check that `isActive`/`startQuery` hasn't been called again -- i.e., the user typed another character in their query. There's boilerplate to do this. At the beginning of the function, we [create an empty object and assign it to a class variable](https://github.com/mozilla-extensions/firefox-quick-suggest-weather/blob/master/src/background.js#L226-L227). Then, after async calls, we [check that the two are still the same](https://github.com/mozilla-extensions/firefox-quick-suggest-weather/blob/master/src/background.js#L262-L264). Could you please add that after these two calls?